http/query.go

```diff
@@ -330,23 +332,27 @@ func QueryRequestFromProxyRequest(req *query.ProxyRequest) (*QueryRequest, error
 
 func decodeQueryRequest(ctx context.Context, r *http.Request, svc platform.OrganizationService) (*QueryRequest, error) {
 	var req QueryRequest
-	// TODO(desa): I'm not sure I like this kind of conditional logic, but it feels better than
-	// introducing another method that does this exact thing.
-	if r.Method == http.MethodGet {
-		qp := r.URL.Query()
-		req.Query = qp.Get("query")
-		if req.Query == "" {
-			return nil, errors.New("query param \"query\" is required")
+
+	ct := r.Header.Get("Content-Type")
+	mt, _, err := mime.ParseMediaType(ct)
+	if err != nil {
+		return nil, err
+	}
+	switch mt {
+	case "application/vnd.flux":
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err
 		}
-	} else {
+		req.Query = string(body)
+	default:
```

Comment:
minor nit... and only for "documentation" purposes, but, do you mind adding
a 

```case "application/json":
      fallthrough
```

to send it to default?